test/integration/select_test.lua

```diff
@@ -77,6 +78,54 @@ pgroup.test_not_valid_value_type = function(g)
     t.assert_str_contains(err.err, "Supplied key type of part 0 does not match index part type: expected unsigned")
 end
 
+pgroup.test_select_nil = function(g)
+    local space = 'customers'
+    local crit_log = "C> empty or nil `select` call on user space=" .. space
+    local valid_condition = {'>=', 'id', 2}
+    local tests = {
+        {crit = true, err = false, uc = nil, opts = nil},
+        {crit = true, err = false, uc = nil, opts = {}},
+        {crit = true, err = false, uc = nil, opts = {fullscan = false}},
+        {crit = true, err = false, uc = {}, opts = nil},
+        {crit = true, err = false, uc = {}, opts = {}},
+        {crit = true, err = false, uc = {}, opts = {fullscan = false}},
+        {crit = false, err = false, uc = nil, opts = {fullscan = true}},
+        {crit = false, err = false, uc = {}, opts = {fullscan = true}},
+        {crit = false, err = true, uc = {0}, opts = nil},
+        {crit = false, err = true, uc = {0}, opts = {}},
+        {crit = false, err = true, uc = {0}, opts = {fullscan = false}},
+        {crit = false, err = false, uc = {valid_condition}, opts = nil},
+        {crit = false, err = false, uc = {valid_condition}, opts = {fullscan = false}},
+        {crit = false, err = false, uc = {valid_condition}, opts = {fullscan = true}},
+    }
+
+    local capture = luatest_capture:new()
+    capture:enable()
+    for _, test in ipairs(tests) do
+        local _, err = g.cluster.main_server.net_box:call('crud.select', {space, test.uc, test.opts})
+        if not test.err then
+            t.assert_equals(err, nil)
+        end
+
+        -- We have a delay here. This hack helps to wait for the end of the output.
+        -- It shouldn't take much time.
+        g.cluster:server('router').net_box:eval([[
+            require('log').error("end of test_select_nil case")
+        ]])
+        local captured = ""
+        while not string.find(captured, "end of test_select_nil case", 1, true) do
```

Comment:
Oh, that hack is really cool! We definitely should use it in similar cases when needed.

Reply:
I extracted it to a separate function.